Fix for charset issue when getting data from TXXX frame.

curtismenmuir opened the pull request:
- userDefinedData was being returned still encoded 
  (E.G. `"data":{"user_description":"LICENCE TYPE","data":"ÿþO"}}`
- Added missing charset into getStringWithCharsetAt() function
- Test updates

aadsm commented:
Thank you so much for looking into this and putting this up! I’ll take a look over the weekend.

aadsm commented:
Nice catch! Can you just fix the indentation? The project uses 2 spaces per indent and this diff changed all the touched files to 4 spaces.

curtismenmuir commented:
@aadsm sorry I was supposed to go back and fix this. Thanks for taking the time to review this! :) I have pushed update to fix this indentation

curtismenmuir commented:
@aadsm is the formatting ok now for getting this merged?

aadsm commented on src/__tests__/ID3v2FrameReader-test.js:
```diff
@@ -338,13 +338,13 @@ describe("ID3v2FrameReader", function() {
-        [0x00, 0x00], // text encoding + null terminated string
+      [0x00, 0x00, 0x00, 0x02], // size
+      [0x00, 0x00], // flags
+      [0x00, 0x00], // text encoding + null terminated string
```
I actually had these indent so you could visually see they belong to the TIT1.

curtismenmuir replied:
Will add space back in

curtismenmuir commented:
@aadsm thanks for reviewing. No worries about taking some time! :) I have removed package-lock.json updates and I have fixed the formatting in the tests!

curtismenmuir commented:
@aadsm do those updates look ok now?

curtismenmuir commented:
@aadsm thanks for merging this :) When are you planning on releasing a new version with the fix? I have a release coming at the end of the month for a project I am working on and it would be great to have the fix included.

aadsm commented:
@curtismenmuir no problem, I'll do that over the weekend.

curtismenmuir commented:
@aadsm that would be perfect, thank you so much!

aadsm commented:
@curtismenmuir fyi: I've just published a new version with your fixes :) I also added you to the contributors list: https://github.com/aadsm/jsmediatags/blob/master/CONTRIBUTORS.md#bugs.
